File: server/DB/models/User.model.js
import { Schema, model } from "mongoose";
import bcrypt from "bcryptjs";
import { appRoles } from "../../src/utils/appRoles.js";

const userSchema = new Schema(
    {
        name: {
            type: String,
            required: true,
            lowercase: true
        },

        email: {
            type: String,
            required: true,
            unique: true,
            lowercase: true
        },

        password: {
            type: String,
            required: true
        },

        phone: {
            type: String,
            required: true
        },

        profileImage: {
            public_id: {
                type: String
            },
            secure_url: {
                type: String
            }
        },

        role: {
            type: String,
            enum: [appRoles.user, appRoles.admin],
            default: appRoles.admin
        },

        isVerified: {
            type: Boolean,
            default: false
        },

        verificationCode: String
    },
    {
        timestamps: true
    }
)
.pre("save", function () {
    this.password = bcrypt.hashSync(this.password, +process.env.SALT_ROUNDS)
})

const userModel = model("User", userSchema)
export default userModel

File: Restaurant/Core/Home/Models/UserModel.swift
//
//  UserModel.swift
//  Restaurant
//
//  Created by Mahmoud Nagdy on 20/01/2026.
//

import Foundation

struct UserModel: Codable {
    let id: String
    let name: String
    let email: String
    let password: String
    let phone: String
    let role: String
    let profileImage: UserImage?
    let isVerified: Bool
    
    
    enum CodingKeys: String, CodingKey {
        case id = "_id", name, email, password, phone, role, profileImage, isVerified
    }
    
}

struct UserImage: Codable {
    let public_id: String
    let secure_url: String
}


struct UserResponseModel: Codable {
    let user: UserModel?
    let message: String
}
